### PR TITLE
Ransomware README with configurable directories

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do? 

Fixes first task of #1287 . If we don't specify a directory, the readme should not be left.

image

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing? 
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...} 
* [x] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: monkey/monkey_island/cc/services/config_schema/ransomware.py [hunk 1]
```diff
@@ -47,7 +47,14 @@
...
                             "directory is specified, no files will be encrypted.",
                         },
+
                     },
+
+                },
+                "readme_note": {
+                        "title": "",
+                        "type": "object",
+                        "description": "A README.txt file will be left alongside the encrypted files.",
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/config_schema/ransomware.py:57]: Can you add a screenshot that shows how this change renders?

File: monkey/tests/unit_tests/infection_monkey/ransomware/test_ransomware_payload.py [hunk 2]
```diff
@@ -244,3 +244,16 @@ def test_readme_already_exists(
...
+    build_ransomware_payload, ransomware_payload_config, ransomware_target
+):
+    ransomware_payload_config["encryption"]["enabled"] = True
+    ransomware_payload_config["encryption"]["directories"]["linux_target_dir"] = ""
+    ransomware_payload_config["encryption"]["directories"]["windows_target_dir"] = ""
+    ransomware_payload_config["other_behaviors"]["readme"] = True
+    ransomware_payload = build_ransomware_payload(ransomware_payload_config)
+
+    ransomware_payload.run_payload()
+    assert not Path(ransomware_target / README_DEST).exists()
```
Comment (mssalvatore) [inline monkey/tests/unit_tests/infection_monkey/ransomware/test_ransomware_payload.py:259]: With an empty target directory, there's no reason to think that a readme would be left in `ransomware_target/`. From the test's perspective, it's very unpredictable where RansomwarePayload might leave the file. It depends on how we're processing the target directory and how that code handles an empty path. This `assert` is not sufficient to verify that no readme file is left on the system.

Take a look at `test_readme_already_exists()`. This test needs to basically do the same thing, which is to assert that no attempt at copying the file is made.

File: monkey/infection_monkey/ransomware/ransomware_payload.py [hunk 3]
```diff
@@ -61,7 +61,8 @@ def run_payload(self):
             file_list = self._find_files()
             self._encrypt_files(file_list)
 
-        self._leave_readme()
+        if self._target_dir:
```
Comment (mssalvatore) [inline monkey/infection_monkey/ransomware/ransomware_payload.py:64]: There's another `if` check on line 95 that also determines whether or not to leave a README. These should both be in the same place. I suggest this:

```python
    def run_payload(self):
        if not self._target_dir:
            return

        LOG.info("Running ransomware payload")

        if self._encryption_enabled:
            file_list = self._find_files()
            self._encrypt_files(file_list)

        if self._readme_enabled:
            self._leave_readme()
```

Then you can remove the first check inside `_leave_readme()` on line 95.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/services/config_schema/ransomware.py (hunk 1 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/services/config_schema/ransomware.py:57]: image

File: monkey/monkey_island/cc/ui/src/components/configuration-components/UiSchema.js [hunk 4]
```diff
@@ -75,8 +75,13 @@ export default function UiSchema(props) {
       encryption: {
         directories: {
             // Directory inputs are dynamically hidden
+        },
+      enabled: {
+
+        'ui:widget': 'hidden'
         }
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/configuration-components/UiSchema.js:82]: This can be put in one line.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**